src/core/contracts/identity.ts

```diff
@@ -128,40 +141,34 @@ export const createAndRegisterBeaconProxy = async (
 };
 
 const getIdentityLogicContractAddress = async (opts?: ConfigOpts): Promise<EthereumAddress> => {
-  const { identityLogic } = getContracts(opts);
   const provider = requireGetProvider(opts);
-
-  const address = identityLogic || (await getContractAddress(provider, IDENTITY_CONTRACT));
-
-  if (!address) throw MissingContract;
+  const address = await getContractAddress(provider, IDENTITY_CONTRACT, opts);
   return address;
 };
 
 const getIdentityCloneFactoryContract = async (opts?: ConfigOpts): Promise<IdentityCloneFactory> => {
-  const { identityCloneFactory } = getContracts(opts);
   const signer = requireGetSigner(opts);
   const provider = requireGetProvider(opts);
-
-  const address = identityCloneFactory || (await getContractAddress(provider, IDENTITY_CLONE_FACTORY_CONTRACT));
-  if (!address) throw MissingContract;
+  const address = await getContractAddress(provider, IDENTITY_CLONE_FACTORY_CONTRACT, opts);
 
   return IdentityCloneFactory__factory.connect(address, signer);
 };
 
 const getBeaconFactoryContract = async (opts?: ConfigOpts): Promise<BeaconFactory> => {
-  const { beaconFactory } = getContracts(opts);
   const signer = requireGetSigner(opts);
   const provider = requireGetProvider(opts);
-
-  const address = beaconFactory || (await getContractAddress(provider, BEACON_FACTORY_CONTRACT));
-  if (!address) throw MissingContract;
```

Comment:
So this pattern turns out to be really confusing for the end user. The `MissingContract` error message tells the user that they need to set their contract addresses in the config, but it's thrown here when we can't find our own contracts on the chain. We were doing this all over the contracts module.

I think we should make a point to be more clear about error naming in the future, so we don't wind up throwing the wrong error message for a problem again.